### PR TITLE
ROX-13963: Third time the charm for showing banner when DB unavailable

### PR DESCRIPTION
**Pull request by vjwilson**
## Description

Take 3 on showing a warning banner in the UI when the Postgres-compatible DB is unavailable.

This version uses the new /database-status endpoint to get its info.

This version incorporates some of Mark's suggestions from an earlier PR, too.

Note: this is similar to #4369,  I had to open another branch with cherry-picked commits, because the last branch got in a bind with merge conflicts.

## Checklist
- [ ] Investigated and inspected CI test results

## Testing Performed

Blocking the endpoint to test the banner:
Screen Shot 2023-01-11 at 5 56 53 PM

And then unblocking the endpoint, and waiting no more than 60 seconds for the API call to poll:
Screen Shot 2023-01-11 at 5 57 54 PM

If the whole server is unreachable, I do not bother to show this banner.
Screen Shot 2023-01-13 at 1 59 41 PM

### CODE REVIEW

File: ui/apps/platform/src/services/DatabaseService.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { DatabaseStatus } from 'types/databaseService.proto';
+
+import axios from './instance';
+
+const databaseUrl = '/v1/database/status';
+
+/**
+ * Fetches database.
+ * TODO return Promise<DatabaseStatus> when component calls directly instead of indirectly via saga.
```
Comment (pedrottimark) [inline ui/apps/platform/src/services/DatabaseService.ts:9]: Is this TODO left over from copy paste?

**Review by pedrottimark — APPROVED** [review]

File: ui/apps/platform/src/types/databaseService.proto.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export type DatabaseStatus = {
+    databaseAvailable: boolean;
+    databaseType?: string | null;
+    databaseVersion?: string | null;
```
Comment (pedrottimark) [inline ui/apps/platform/src/types/databaseService.proto.ts:4]: Although `?` optional is needed, `| null` is not also needed based on reply from David Shrewsberry in https://github.com/stackrox/stackrox/pull/4369/files#r1068572917

> I don't think you will receive a null unless your (java|type)script interprets an empty string as such. Here is what a curl will provide you:

```js
{"databaseAvailable":false,"databaseType":"Hidden","databaseVersion":""}
```

notice the databaseVersion is an empty string.

**Review by dashrews78 — APPROVED** [review]
LGTM.  Sorry it became a pain

File: ui/apps/platform/src/Containers/MainPage/MainPage.tsx [hunk 3]
```diff
@@ -98,6 +99,7 @@ function MainPage(): ReactElement {
                     hasServiceIdentityWritePermission={hasServiceIdentityWritePermission}
                 />
                 {metadata?.stale && <VersionOutOfDate />}
+                 <DatabaseBanner isApiReachable={serverState && serverState !== 'UNREACHABLE'} />
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/MainPage/MainPage.tsx:102]: style-checks reports an error for extra space at the left of `<DatabaseBanner` directly under `{` above

**Pull request merged**